src/d3a/models/strategy/storage.py

```diff
@@ -226,15 +231,45 @@ def event_tick(self, *, area):
     def event_trade(self, *, market_id, trade):
         market = self.area.get_future_market_from_id(market_id)
         super().event_trade(market_id=market_id, trade=trade)
+        if trade.buyer == self.owner.name:
+            self._energy_bought_type(trade)
         if trade.offer.seller == self.owner.name:
+            self._energy_sell_type(trade)
             self.state.pledged_sell_kWh[market.time_slot] += trade.offer.energy
             self.state.offered_sell_kWh[market.time_slot] -= trade.offer.energy
 
+    def _is_local(self, trade):
+        for child in self.area.children:
+            if child.name == trade.seller:
+                return True
+
+    def _energy_sell_type(self, trade):
```

Comment:
As this is only for tracking, please rename e.g. to `_track_energy_sell_type`

Reply:
The same for `_energy_bought_type`

Reply:
Also, please leave a comment here, that the FIRST-IN FIRST-OUT mechanism is used (just for later). Thanks!